### PR TITLE
Modify verify script to use microservice

### PR DESCRIPTION
**Pull request by alextserepov**

### CODE REVIEW

File: py/sigver/verify.py [hunk 1]
```diff
@@ -3,24 +3,20 @@
...
 from sha256tree import sha256sum
+import requests
+import json
+import base64
 
-KEY_VAULT_URL = "https://ghaf-devenv-ca.vault.azure.net/"
-CERTIFICATE_NAME = "."
-
+CERTIFICATE_NAME="INT-Ghaf-Devenv-Common"
+url = "https://ghaf-devenv-microsign-aleksandrtserepo-app.azurewebsites.net/api/verify-signature"
```
Comment (joinemm) [inline py/sigver/verify.py:12]: Is the plan to move this to some better url?

Comment (alextserepov) [reply py/sigver/verify.py:12]: Yes, sure. Very good point.

**Review by joinemm — APPROVED** [review]

**Comment by henrirosten** [comment]
Consider fixing pylint warnings you see relevant:
```
❯ pylint py/sigver/verify.py 
************* Module verify
py/sigver/verify.py:1:0: C0114: Missing module docstring (missing-module-docstring)
py/sigver/verify.py:14:0: C0116: Missing function or method docstring (missing-function-docstring)
py/sigver/verify.py:25:0: C0116: Missing function or method docstring (missing-function-docstring)
py/sigver/verify.py:74:11: W0718: Catching too general exception Exception (broad-exception-caught)
py/sigver/verify.py:8:0: C0411: standard import "json" should be placed before third party imports "sha256tree.sha256sum", "requests" (wrong-import-order)
py/sigver/verify.py:9:0: C0411: standard import "base64" should be placed before third party imports "sha256tree.sha256sum", "requests" (wrong-import-order)
```

Other improvements I think we could do in this PR already:
- Consider returning exit status if an error occurred or if signature was invalid. This would make it far easier to integrate the `verify` to other scripts (versus parsing the textual output)
- Consider adding some basic sanity checks for the signature (e.g. file size?). As it is now, one can accidentally do something like `nix run .#verify -- --path=image.img.sig --sigfile=image.img` (that is, to accidentally mix the arguments), and the script would happily dump the base64 encoded img file to the console and attempt to send it to Azure.

Other improvements you could also think of doing (perhaps in a follow-up PRs later):
- Use `argparse` to parse the command line arguments, rather than parsing them manually
- Use  `logging` to log rather than `print`
- Add pylint checks etc. as github actions checks so they would be run automatically for all PRs

File: py/sigver/verify.py (hunk 1 above)
Comment (cazfi) [reply py/sigver/verify.py:12]: Not a matter of this PR, but be sure to document adjusting this URL, as well as whole setup of this function, in ghaf-infra setup instructions.

Comment (alextserepov) [reply py/sigver/verify.py]: sure. good point.

**Comment by alextserepov** [comment]
> Consider fixing pylint warnings you see relevant:
> 
> ```
> ❯ pylint py/sigver/verify.py 
> ************* Module verify
> py/sigver/verify.py:1:0: C0114: Missing module docstring (missing-module-docstring)
> py/sigver/verify.py:14:0: C0116: Missing function or method docstring (missing-function-docstring)
> py/sigver/verify.py:25:0: C0116: Missing function or method docstring (missing-function-docstring)
> py/sigver/verify.py:74:11: W0718: Catching too general exception Exception (broad-exception-caught)
> py/sigver/verify.py:8:0: C0411: standard import "json" should be placed before third party imports "sha256tree.sha256sum", "requests" (wrong-import-order)
> py/sigver/verify.py:9:0: C0411: standard import "base64" should be placed before third party imports "sha256tree.sha256sum", "requests" (wrong-import-order)
> ```
> 
> Other improvements I think we could do in this PR already:
> 
>     * Consider returning exit status if an error occurred or if signature was invalid. This would make it far easier to integrate the `verify` to other scripts (versus parsing the textual output)
> 
>     * Consider adding some basic sanity checks for the signature (e.g. file size?). As it is now, one can accidentally do something like `nix run .#verify -- --path=image.img.sig --sigfile=image.img` (that is, to accidentally mix the arguments), and the script would happily dump the base64 encoded img file to the console and attempt to send it to Azure.
> 
> 
> Other improvements you could also think of doing (perhaps in a follow-up PRs later):
> 
>     * Use `argparse` to parse the command line arguments, rather than parsing them manually
> 
>     * Use  `logging` to log rather than `print`
> 
>     * Add pylint checks etc. as github actions checks so they would be run automatically for all PRs

Almost there.

**Review by cazfi — APPROVED** [review]

File: py/sigver/verify.py [hunk 2]
```diff
@@ -1,85 +1,88 @@
...
-    print(certificate.name)
+    with open(sigfile, "rb") as file:
+        sig = file.read()
 
-    key = key_client.get_key(certificate_name)
+    signature = base64.b64encode(sig).decode('utf-8')
 
-    crypto_client = CryptographyClient(key, credential)
+    data = {
+        "certificateName": certificate_name,
```
Comment (henrirosten) [inline py/sigver/verify.py:66]: Consider removing trailing space on line 66

File: py/sigver/verify.py [hunk 3]
```diff
@@ -1,85 +1,88 @@
...
 
-        args.pop(0)
+    path=args.path
+    certificate_name = args.cert
+    sigfile = args.sigfile
 
-    credential = DefaultAzureCredential()
+    digest = base64.b64encode(sha256sum(path, 1024 * 1024, True)).decode('utf-8')
+    if os.path.getsize(sigfile) != 64:
+        LOG.info("Wrong signature size!")
```
Comment (henrirosten) [inline py/sigver/verify.py:57]: Should this be `LOG.error` instead?

File: py/sigver/verify.py [hunk 4]
```diff
@@ -1,85 +1,88 @@
...
-    crypto_client = CryptographyClient(key, credential)
+    data = {
+        "certificateName": certificate_name, 
+        "Hash": digest,
+        "Signature": signature
+    }
 
-    digest = sha256sum(path, 1024 * 1024, True)
+    headers = {"Content-Type": "application/json"}
+    LOG.info (json.dumps(data))
```
Comment (henrirosten) [inline py/sigver/verify.py:72]: Default logging level is WARNING, meaning e.g. LOG.info(...) will never be logged as it now is.
Typically, one would provide an argument (e.g. `-v/--verbose`) to increase the logging verbosity.

Consider adding `--verbose` maybe in a follow-up.

File: py/sigver/verify.py [hunk 5]
```diff
@@ -1,85 +1,88 @@
...
+
 
+# Default certificate to be used if no argument is given
+CERTIFICATE_NAME="INT-Ghaf-Devenv-Common"
+
+# Azure Function (verify-signature) URL
+URL = "https://ghaf-devenv-microsign-aleksandrtserepo-app.azurewebsites.net/api/verify-signature"
 
 def show_help():
+    """Show help messsage"""
```
Comment (henrirosten) [inline py/sigver/verify.py:27]: Note: this function `show_help(...)` is no longer needed after you moved to using `argparse`.

**Pull request merged**